src/libraries/System.Text.RegularExpressions/tests/Regex.Groups.Tests.cs

```diff
@@ -859,6 +859,9 @@ public static IEnumerable<object[]> Groups_Basic_TestData()
                 yield return new object[] { engine, null, @"(?(cat)|catdog)", "catdog", RegexOptions.None, new string[] { "" } };
                 yield return new object[] { engine, null, @"(?(cat)|dog)", "dog", RegexOptions.None, new string[] { "dog" } };
 
+                yield return new object[] { engine, null, @"(?((\w{3}))\1\1|no)", "dogdogdog", RegexOptions.None, new string[] { "dogdog", "dog" } };
```

Comment:
do we have sufficient tests in general that we appropriately advance after this match? Looks like these tests just get the first match. What if the code you changed did not advance correctly after that?